### PR TITLE
Adds a chemistry/missing page for curation

### PR DESCRIPTION
**Pull request by egonw**
- with column on the `/chemical/` frontpage
- adds a few more example chemical structures 

I expect more patches later for more physchem properties from @lahire.

### CODE REVIEW

**Comment by egonw** [comment]
Implements https://github.com/fnielsen/scholia/issues/546

**Comment by fnielsen** [comment]
1. Headline level should be h2. May be a upper headline as h1 with "Missing information for chemical" or something. There is also a bit too much whitespace 
2. "sparqlToDataTable(missingDensitySparql, "#missing-density");" has too much whitespace. A couple or lines down also have too much whitespace.
3. Should the name of the template be different, e.g., chemical_empty_missing. The aspect_missing filenames are usually the ones pertaining to a specific Q-item, e.g., /scholia/chemical/Q123/missing. 
4. Perhaps we should be consistent with the case for the named subqueries. I hvae tended to use lowercase. I see that https://wiki.blazegraph.com/wiki/index.php/NamedSubquery uses lowercase+camelCase

**Comment by Daniel-Mietchen** [comment]
Some more observations:
- the current queries do not seem to prioritize curation in any way (e.g. via sitelinks) but that would probably be helpful
- each of these tables is an opportunity to teach about the underlying concept, so "Missing Boiling Points" could link to [boiling point (Q1003183)](https://www.wikidata.org/wiki/Q1003183) and perhaps provide some [usage examples](https://www.wikidata.org/wiki/Property:P2102#P1855)
- some generic help link regarding how to handle units may also be useful
- typo: "Enthropy"

**Comment by egonw** [comment]
@fnielsen, I suggest we make your point 3 and 4 issues, and do that later. (Done for point 3 already).

@Daniel-Mietchen, I like the idea for using sitelinks. I'm on it.  Also +1 on the documentation/guidance. Excellent point.

**Comment by egonw** [comment]
hi @Daniel-Mietchen, @fnielsen, I tried to split nice-to-have from must-have-before-merge. Please have a look at the new patches, and let me know if there are further must-have-before-merge's.

File: scholia/app/templates/chemicals_missing.html [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+  <li>use references: <a href="https://www.wikidata.org/wiki/Help:Sources">Help:Sources</a></li>
+</ul>
+
+The following links prioritize chemical compounds that are listed in most Wikis
+(Wikipedias, WikiCommons, etc). To go to the Wikidata page for each item, click the name
+below and then on the shown Scholia page for that compound the Q-identifier link at the top.
+
+<h2 id="h1">Missing Boiling Points</h2>
+
+The <a href="https://www.wikidata.org/wiki/Property:P2102">boiling point</a> is the temperature above
```
Comment (Daniel-Mietchen) [inline scholia/app/templates/chemicals_missing.html]: That's not the definition of boiling point.

File: scholia/app/templates/chemicals_missing.html [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+which the compound is liquid. Typical units given include Kelvin and degree Celsius. Because it
```
Comment (Daniel-Mietchen) [inline scholia/app/templates/chemicals_missing.html]: Opportunity to link to "temperature" or "Kelvin" and "Celsius".

File: scholia/app/templates/chemicals_missing.html [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+is depends on the pressure, this is often also given, as shown in
```
Comment (Daniel-Mietchen) [inline scholia/app/templates/chemicals_missing.html]: Typo in "it is depends".

File: scholia/app/templates/chemicals_missing.html [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+<table class="table table-hover" id="missing-pKa"></table>
+
+<h2 id="h1">Missing Combustion Enthalpy</h2>
+
+<table class="table table-hover" id="missing-combustion-enthalpy"></table>
+
+<h2 id="h1">Missing IDLH</h2>
+
+The <a href="https://www.wikidata.org/wiki/Property:P2129">Immediately Dangerous to Life and Health value</a>
+(IDLH) is an indication provided by the U.S.A. CDC about the danger of the compound.
```
Comment (Daniel-Mietchen) [inline scholia/app/templates/chemicals_missing.html]: Opportunity to link to a CDC profile, perhaps https://tools.wmflabs.org/scholia/organization/Q583725 .

File: scholia/app/templates/chemicals_missing.html [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+     sparqlToDataTable(missingCombustionEnthalpy, "#missing-combustion-enthalpy");
+     sparqlToDataTable(missingIDLH, "#missing-idlh");
+ });
+</script>
+ 
+{% endblock %}
+
+{% block page_content %}
+
+<h1>Missing physicalchemical properties for chemicals</h1>
```
Comment (Daniel-Mietchen) [inline scholia/app/templates/chemicals_missing.html]: physicochemical

**Comment by Daniel-Mietchen** [comment]
I left a few more comments in the code &mdash; nothing serious, and easy to fix. I also tried to load the missing page multiple times, and there was almost always some table that did not load, presumably due to the rate limit, as per #302 .

File: scholia/app/templates/chemicals_missing.html (hunk 1 above)
Comment (egonw) [reply scholia/app/templates/chemicals_missing.html]: Fixed.

File: scholia/app/templates/chemicals_missing.html (hunk 2 above)
Comment (egonw) [reply scholia/app/templates/chemicals_missing.html]: Linked to Kelvin and Celsius (and linked literature to both of them).

File: scholia/app/templates/chemicals_missing.html (hunk 3 above)
Comment (egonw) [reply scholia/app/templates/chemicals_missing.html]: Fixed.

File: scholia/app/templates/chemicals_missing.html (hunk 4 above)
Comment (egonw) [reply scholia/app/templates/chemicals_missing.html]: Added

File: scholia/app/templates/chemicals_missing.html (hunk 5 above)
Comment (egonw) [reply scholia/app/templates/chemicals_missing.html]: Fixed.

**Comment by egonw** [comment]
Regarding the rate limit, I note too that they queries have become quite slow. This may have to do that Wikidata now has >1M compounds, instead of the 180k when these queries were made.

**Comment by egonw** [comment]
It is the `wikibase:sitelinks` that makes it slow, it seems: https://phabricator.wikimedia.org/T175380

**Comment by egonw** [comment]
Okay, the version now is a lot faster, BUT also returns less data. Particularly for more common properties this may be a problem, as it now only looks at compounds with at least 40 sitelinks (30 for molecular mass).

**Comment by Daniel-Mietchen** [comment]
I think this is good to go now, even though there is also room for improvements, e.g. I think we should look into integrating with TABernacle for making the actual edits. Not sure it can do qualifiers though.

**Comment by egonw** [comment]
`tox -e flake8` passed.

**Pull request merged**